### ISSUE TITLE
Selector-no-type complains about double slash comments

### ISSUE THREAD

**Issue by TheSisb** [stylelint/stylelint#2605]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug in `selector-no-type` that complains on double slash comments.

> Which rule, if any, is this issue related to?

`selector-no-type`

> What CSS is needed to reproduce this issue?

e.g.

```css
// test
.foo {
  color: pink;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    'selector-no-type': [true, {ignore: 'descendant'}]
  }
}
```

> Which version of stylelint are you using?

`7.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config myconfig *.css`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, we use postcss to allow inline comments (https://www.npmjs.com/package/postcss-comment)

> What did you expect to happen?

No warnings to be flagged

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
1:1  ✖  Unexpected type selector   selector-no-type
```

**Comment by hudochenkov** [comment]
Thank you for reporting and using template!

It will be helpful if error message was for example stylesheet you provided :)

The problem is in `// comments`. It's not a standard CSS, but you're using `.css` file extensions. stylelint can't parse it correctly. I think if you set syntax to `scss` it will fix your problem. stylelint will use SCSS parser instead, and this parser will correctly parse your `// comments`.

```
stylelint --config myconfig --syntax scss *.css
```

**Comment by TheSisb** [comment]
That fixes the issue, thanks!

Is there a way to set the syntax from the `stylint.config.js` file?  I couldn't find it in the docs.

**Comment by hudochenkov** [comment]
No, it isn't possible.

**Comment by TheSisb** [comment]
That's kind of problematic.  Webpack and my IDE now still complain about the issue.  :/

**Comment by jeddy3** [comment]
Can you change the extension of your files to `.scss`? If your writing SCSS rather than CSS, then you ought to use the corresponding file extension. I believe both your IDE and webpack will then detect the`.scss` extension and set the syntax to SCSS.

**Comment by TheSisb** [comment]
We're currently using postcss with css modules to parse our stylesheets.  Most of our css is true to the original css, but we really wanted to use inline comments `//` because we migrated from less and they're simply convenient.

Our extensions are `filename.module.css`, which I *think* is the current lead for naming conventions when using css modules.

**Comment by hudochenkov** [comment]
For webpack you can configure syntax: https://github.com/JaKXz/stylelint-webpack-plugin#options

**Comment by jeddy3** [comment]
The standard PostCSS parser [will parse](https://astexplorer.net/#/gist/f7a97f59fcda7a5a87448f178b8021b3/5c6277307c10f9124ca7fb8ba5cc2b57b68d0154) `// test` as part of the selector, as SCSS-like comments are a non-standard construct that require a special parser to be understood.

At some point in the configuration of your tools (be it webpack or your IDE) you'll need to specify that a [special PostCSS parser](https://github.com/postcss/postcss#syntaxes) should be used because your using a non-standard construct in a file with a `.css` extension.

As @hudochenkov said, there's an option in `stylelint-webpack-plugin` to do that and in stylelint itself there are the `--syntax` and `--custom-syntax` options. There may or may not be an option in your IDE to do it.

As the syntax is often inferred from the file extension by tools (including stylelint), I'd recommend (as it'll likely save you a bunch of trouble) either using `filename.module.scss` if you're going to use SCSS-like comments, or [drop the SCSS-like comments](http://davidtheclark.com/lessons-i-may-have-learned/#non-standard-syntaxes-are-a-huge-obstacle-to-css-tooling) and keep the `.css` extension.

As the `selector-no-type` rule is behaving as expected, I think we can close this issue?

**Comment by TheSisb** [comment]
Thank you very much for the clarification and help!
